### ISSUE TITLE
NATS Transporter: unhandled promise rejection upon disconnect

### ISSUE THREAD

**Issue by deividasjackus** [ice-services/moleculer#62]
Upon sending `SIGINT` to an application, the NATS transporter sends a disconnect packet, every now and then resulting in an unhandled promise rejection:

```
Unhandled rejection TypeError: Cannot read property 'publish' of null
    at NatsTransporter.publish (\node_modules\moleculer\src\transporters\nats.js:151:14)
    at Transit.publish (\node_modules\moleculer\src\transit.js:494:18)
    at Transit.sendDisconnectPacket (\node_modules\moleculer\src\transit.js:162:8)
    at Transit.disconnect (\node_modules\moleculer\src\transit.js:145:16)
    at Promise.resolve.then.catch.then (\node_modules\moleculer\src\service-broker.js:291:26)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
From previous event:
    at ServiceBroker.stop (\node_modules\moleculer\src\service-broker.js:289:5)
    at process.ServiceBroker._closeFn (\node_modules\moleculer\src\service-broker.js:159:9)
    at emitOne (events.js:115:13)
    at process.emit (events.js:210:7)
```

**Comment by deividasjackus** [comment]
@icebob I have not tested this with other transporters. Based on the documentation I assume that both Redis and MQTT transporters might have been based on the NATS transporter.

**Comment by icebob** [comment]
Thanks, I will check it.

**Comment by icebob** [comment]
Released in v0.8.5
